packets.go

```diff
@@ -179,8 +179,11 @@ func (mc *mysqlConn) readInitPacket() ([]byte, error) {
 	}
 
 	// server version [null terminated string]
+	idPos := 1 + bytes.IndexByte(data[1:], 0x00) + 1
+	mc.id = int(binary.LittleEndian.Uint32(data[idPos : idPos+4]))
+
```

Comment:
These two lines should be after `// connection id [4 bytes]` comment.

Reply:
Agreee